anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -726,16 +726,16 @@ def _find_match(
         :return: sqlalchemy query object
         :rtype: Query
         """
-        results = db.query(GrypeDBMetadata)
+        results = db.query(GrypeDBFeedMetadata)
         if checksum:
-            results = results.filter(GrypeDBMetadata.checksum == checksum)
+            results = results.filter(GrypeDBFeedMetadata.archive_checksum == checksum)
         if not isinstance(active, type(None)):
-            results = results.filter(GrypeDBMetadata.active == active)
+            results = results.filter(GrypeDBFeedMetadata.active == active)
         return results
```

Comment:
Return the results here, not the query. Operate on the returned objects using the session as necessary. Details in the below comment